File: proyecto-mds2/frontend/vista-iniciar_sesion.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/vaadin-text-field/src/vaadin-password-field.js';

@customElement('vista-iniciar_sesion')
export class VistaIniciar_sesion extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-vertical-layout style="width: 100%;" id="vertical">
 <vaadin-vertical-layout theme="spacing" style="align-self: center; height: 100%; padding: var(--lumo-space-xl);" id="verticalContenedor">
  <vaadin-horizontal-layout theme="spacing" style="align-self: center;" id="horizontalTitulo">
   <label style="flex-shrink: 1; flex-grow: 0; font-size: 25px" id="labelTitulo">Iniciar Sesión</label>
  </vaadin-horizontal-layout>
  <vaadin-horizontal-layout theme="spacing-xl" style="width: 100%; height: 100%; align-items: center;" id="vaadinHorizontalLayout2">
   <label style="width: 100%;" id="labelUsuario">Nombre de usuario</label>
   <vaadin-text-field error-message="Introduce un valor" invalid="" style="flex-grow: 0; flex-shrink: 0; margin-top: var(--lumo-space-m);" placeholder="Usuario" id="tfUsuario" required></vaadin-text-field>
  </vaadin-horizontal-layout>
  <vaadin-horizontal-layout theme="spacing-xl" style="width: 100%; height: 100%; align-items: center;" id="horizontalPass">
   <label style="flex-shrink: 1; flex-grow: 0; width: 100%;" id="labelPass">Contraseña</label>
   <vaadin-password-field label="" placeholder="Contraseña" value="" id="tfPass"></vaadin-password-field>
  </vaadin-horizontal-layout>
  <vaadin-horizontal-layout theme="spacing-xl" style="align-self: flex-start; width: 100%; height: 50px;" id="horizontalLogin">
   <img style="height: 100%;" src="https://cdn-icons-png.flaticon.com/512/2991/2991148.png" id="imgGoogle">
   <img style="height: 100%;" src="https://cdn-icons-png.flaticon.com/512/124/124010.png" id="imgFace">
   <img style="height: 100%;" src="https://upload.wikimedia.org/wikipedia/commons/thumb/f/fa/Apple_logo_black.svg/814px-Apple_logo_black.svg.png" id="imgApple">
   <vaadin-button style="width: 100%; padding: var(--lumo-space-xs);" id="buttonLogin">
     Iniciar Sesión 
   </vaadin-button>
  </vaadin-horizontal-layout>
  <vaadin-horizontal-layout theme="spacing-xl" style="width: 100%;" id="horizontalRecuperar">
   <label style="width: 100%;" id="labelRecuperar">¿Ha olvidado su contraseña?</label>
   <vaadin-button style="width: 100%; padding: var(--lumo-space-xs);" id="botonRecuperar">
     Recuperar cuenta 
   </vaadin-button>
  </vaadin-horizontal-layout>
  <vaadin-vertical-layout theme="spacing" id="vaadinVerticalLayout" style="align-items: center; align-self: center; width: 100%;">
   <label style="align-self: center;" id="labelRegistro">¿No tienes cuenta?</label>
   <vaadin-button style="height: 100%;" id="botonRegistro">
     Regístrate aquí 
   </vaadin-button>
   <vaadin-button id="botonVolver">
     Volver atrás 
   </vaadin-button>
  </vaadin-vertical-layout>
 </vaadin-vertical-layout>
</vaadin-vertical-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: proyecto-mds2/src/main/java/vistas/VistaIniciar_sesion.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.html.Label;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.textfield.PasswordField;

/**
 * A Designer generated component for the vista-iniciar_sesion template.
 *
 * Designer will add and remove fields with @Id mappings but
 * does not overwrite or otherwise change this file.
 */
@Tag("vista-iniciar_sesion")
@JsModule("./vista-iniciar_sesion.ts")
public class VistaIniciar_sesion extends LitTemplate {

	@Id("verticalContenedor")
	private Element verticalContenedor;
	@Id("vertical")
	private Element vertical;
	@Id("horizontalTitulo")
	private HorizontalLayout horizontalTitulo;
	@Id("labelTitulo")
	private Label labelTitulo;
	@Id("tfUsuario")
	private TextField tfUsuario;
	@Id("horizontalPass")
	private HorizontalLayout horizontalPass;
	@Id("labelUsuario")
	private Label labelUsuario;
	@Id("vaadinHorizontalLayout2")
	private HorizontalLayout vaadinHorizontalLayout2;
	@Id("horizontalLogin")
	private HorizontalLayout horizontalLogin;
	@Id("imgGoogle")
	private Image imgGoogle;
	@Id("imgFace")
	private Image imgFace;
	@Id("imgApple")
	private Image imgApple;
	@Id("buttonLogin")
	private Button buttonLogin;
	@Id("horizontalRecuperar")
	private HorizontalLayout horizontalRecuperar;
	@Id("labelRecuperar")
	private Label labelRecuperar;
	@Id("botonRecuperar")
	private Button botonRecuperar;
	@Id("vaadinVerticalLayout")
	private Element vaadinVerticalLayout;
	@Id("labelRegistro")
	private Label labelRegistro;
	@Id("botonRegistro")
	private Button botonRegistro;
	@Id("botonVolver")
	private Button botonVolver;
	@Id("tfPass")
	private PasswordField tfPass;
	@Id("labelPass")
	private Label labelPass;
	public Element getVerticalContenedor() {
		return verticalContenedor;
	}
	public Element getVertical() {
		return vertical;
	}
	public HorizontalLayout getHorizontalTitulo() {
		return horizontalTitulo;
	}
	public Label getLabelTitulo() {
		return labelTitulo;
	}
	public Label getLabelPass() {
		return labelPass;
	}
	public TextField getTfUsuario() {
		return tfUsuario;
	}
	public HorizontalLayout getHorizontalPass() {
		return horizontalPass;
	}
	public Label getLabelUsuario() {
		return labelUsuario;
	}
	public HorizontalLayout getVaadinHorizontalLayout2() {
		return vaadinHorizontalLayout2;
	}
	public HorizontalLayout getHorizontalLogin() {
		return horizontalLogin;
	}
	public Image getImgGoogle() {
		return imgGoogle;
	}
	public Image getImgFace() {
		return imgFace;
	}
	public Image getImgApple() {
		return imgApple;
	}
	public Button getButtonLogin() {
		return buttonLogin;
	}
	public HorizontalLayout getHorizontalRecuperar() {
		return horizontalRecuperar;
	}
	public Label getLabelRecuperar() {
		return labelRecuperar;
	}
	public Button getBotonRecuperar() {
		return botonRecuperar;
	}
	public Element getVaadinVerticalLayout() {
		return vaadinVerticalLayout;
	}
	public Label getLabelRegistro() {
		return labelRegistro;
	}
	public Button getBotonRegistro() {
		return botonRegistro;
	}
	public Button getBotonVolver() {
		return botonVolver;
	}
	public PasswordField getTfPass() {
		return tfPass;
	}
	public void setTfPass(PasswordField tfPass) {
		this.tfPass = tfPass;
	}

	
	

}
